### PR TITLE
Tx Path Unification and Enhancements

### PR DESCRIPTION
**Pull request by PhoenixWorthVCD**
## Description

- Unified the tx partitioning path (txs/txt search) and the default path (txt search only) @ full_loop_core() into 1 unique path (i.e. product_full_loop_tx_search() is removed, and only tx_type_search() is now called). This will make future development(s) easier (1 spot to modify instead of 2).
- Added txs early exit (if cost of performed child tx(s) is higher than the cost of the Parent tx then exit current Tx depth) .
- Bypassed txt search for Tx(s) where only DCT is allowed (if tx width > 32 or tx high > 32).
- Bypassed INTRA compensation for Tx depth 0 as previous stage predicted samples could be re-used.
 - Added the ability to specify whether frequency or spatial domain @ distortion calculation per PD/MD stage.
- Removed useless tx signal(s).  
- Updated tx block geometry.

## Type of Change

Enhancement

## Authors

@hguermaz 
@PhoenixWorthVCD 

## Testing and Performance

The changes are lossless. No speed or memory overhead.

### CODE REVIEW

**Comment by hguermaz** [comment]
@PhoenixWorthVCD after rebasing could you please check again that streams are identical for the subset you are using, then check using the Travis command lines.

File: Source/Lib/Encoder/Codec/EbCodingLoop.c [hunk 1]
```diff
@@ -1547,8 +1566,12 @@ void perform_intra_coding_loop(PictureControlSet *pcs_ptr, SuperBlock *sb_ptr, u
...
                 left_neigh_array + 1,
                 recon_buffer,
-                context_ptr->blk_geom->tx_boff_x[blk_ptr->tx_depth][context_ptr->txb_itr] >> 2,
-                context_ptr->blk_geom->tx_boff_y[blk_ptr->tx_depth][context_ptr->txb_itr] >> 2,
+                (context_ptr->blk_geom->tx_org_x[0][blk_ptr->tx_depth][context_ptr->txb_itr] -
+                 context_ptr->blk_geom->origin_x) >>
+                    2,
+                (context_ptr->blk_geom->tx_org_y[0][blk_ptr->tx_depth][context_ptr->txb_itr] -
+                 context_ptr->blk_geom->origin_y) >>
+                    2,
```
[the remark this answers is not in the document]
Comment (hguermaz) [reply Source/Lib/Encoder/Codec/EbCodingLoop.c]: It makes sense. @PhoenixWorthVCD could you please use is_inter always, and is_inter will be set equal to 0 if INTRA path, and to 1 if INTER path.

[the remark this answers is not in the document]
Comment (hguermaz) [reply Source/Lib/Encoder/Codec/EbCodingLoop.c]: It makes sense. @PhoenixWorthVCD could you please use is_inter always, and is_inter should be set equal to 0 if INTRA path, and to 1 if INTER path.

[the remark this answers is not in the document]
Comment (hguermaz) [reply Source/Lib/Encoder/Codec/EbCodingLoop.c]: It makes sense. @PhoenixWorthVCD could you please use is_inter always, and is_inter should be set equal to 0 if INTRA path, and to 1 if INTER path.

[the remark this answers is not in the document]
Comment (hguermaz) [reply Source/Lib/Encoder/Codec/EbCodingLoop.c]: It makes sense. @PhoenixWorthVCD could you please use is_inter always, and is_inter should be set equal to 0 if INTRA path, and to 1 if INTER path.

[the remark this answers is not in the document]
Comment (hguermaz) [reply Source/Lib/Encoder/Codec/EbCodingLoop.c]: It makes sense. @PhoenixWorthVCD could you please use is_inter always, and is_inter should be set equal to 0 if INTRA path, and to 1 if INTER path.

[the remark this answers is not in the document]
Comment (PhoenixWorthVCD) [reply Source/Lib/Encoder/Codec/EbCodingLoop.c]: done.

**Comment by PhoenixWorthVCD** [comment]
> @PhoenixWorthVCD after rebasing could you please check again that streams are identical for the subset you are using, then check using the Travis command lines.

I verified that the streams are identical on my clip subset.

**Pull request merged**